test/security-test.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+# prevent souring of this script, only allow execution
+$(return >/dev/null 2>&1)
+test "$?" -eq "0" && { echo "This script should only be executed." >&2; exit 1; }
+
+set -x
+
+# set log id and use shared log function for readable logs
+declare mydir
+mydir=$(cd "$(dirname "${BASH_SOURCE[0]}")" &>/dev/null && pwd -P)
+declare HOPR_LOG_ID="e2e-test"
+source "${mydir}/../scripts/utils.sh"
+
+usage() {
+  msg
+  msg "Usage: $0 <host> <rest_port> <admin_port>"
+  msg
+}
+
+declare host="${1}"
+declare rest_port="${2}"
+declare admin_port="${3}"
+
+log "Security tests started"
+log "Rest API @ ${host}:${rest_port}"
+log "WS API @ ${host}:${admin_port}"
+
+# test REST port open
+nc -z ${host} ${rest_port}
+
+# test admin port open
+nc -z ${host} ${admin_port}
+
+# should fail REST authentication without proper token
+log "Testing REST rejecting null token"
```

Comment:
log message is not clear, consider this is being seen by another team member with no context.